twisted/conch/ssh/keys.py

```diff
@@ -802,9 +816,18 @@ def fingerprint(self):
 
         @rtype: L{str}
         """
-        return nativeString(
-            b':'.join([binascii.hexlify(x)
-                       for x in iterbytes(md5(self.blob()).digest())]))
+        hashFunction = {'md5-hex' : md5, 'sha256-base64' : sha256}
+
+        if format == 'sha256-base64':
+            return nativeString(base64.b64encode(
+                hashFunction[format](self.blob()).digest()))
+        elif format == 'md5-hex':
+            return nativeString(
+                b':'.join([binascii.hexlify(x)
+                for x in iterbytes(hashFunction[format](self.blob()).digest())]))
+        else:
+            raise BadFingerPrintFormat(
+                'Unsupported fingerprint format: %s' % (format,))
```

Comment:
So here we have `if` tests on the entire string, as well as a `hashFunction` dict. The problem with that is that the `hashFunction` dict lookup serves no purpose: we have already had to check the string value in the `if` test, so *then* doing a dict lookup is just giving us a hardcoded answer that we already know.

In this case, I think `hashFunction` can go.